ctc_lib/logging/log_manager.go

```diff
@@ -0,0 +1,44 @@
+/*
+Copyright 2018 Google Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package logging
+
+import (
+	"time"
+
+	rotatelogs "github.com/lestrrat-go/file-rotatelogs"
+	log "github.com/sirupsen/logrus"
+)
+
+func NewLogger(path string) *log.Logger {
+	writer, _ := rotatelogs.New(
```

Comment:
I don't see a ton in the documentation for how this works: https://github.com/lestrrat-go/file-rotatelogs

Are we sure it works across invocations of the tool? What do we get out of it instead of generating a new tmpfile in the same directory everytime the tool is run?

Reply:
We get a bunch of stuff.
```
	writer, _ := rotatelogs.New(
		path+".%Y%m%d%H%M",
		rotatelogs.WithLinkName(path),
		rotatelogs.WithMaxAge(time.Duration(86400)*time.Second),
		rotatelogs.WithRotationTime(time.Duration(86400)*time.Second),
	)
```
With the above code, 
1. We are rotating logs daily.
2.  Old logs are purged after a day.

Example:
```
writer, _ := rotatelogs.New(
		path+".%Y%m%d%H%M%S",
		rotatelogs.WithLinkName(path),
		rotatelogs.WithMaxAge(time.Duration(20)*time.Second),
		rotatelogs.WithRotationTime(time.Duration(60)*time.Second),
	)
```
1.First Run
```
$bazel run demo/ctc:ctc_demo --
$ls -al /tmp/echo*
lrwxrwxrwx 1 tejaldesai primarygroup  24 Mar 23 11:27 /tmp/echo -> /tmp/echo.20180323112700
-rw-r--r-- 1 tejaldesai primarygroup 321 Mar 23 11:27 /tmp/echo.20180323112700

```

2. After 20 Seconds, another file written
```
$bazel run demo/ctc:ctc_demo -- 
echo
$ ls -al /tmp/echo*
lrwxrwxrwx 1 tejaldesai primarygroup  24 Mar 23 11:28 /tmp/echo -> /tmp/echo.20180323112800
-rw-r--r-- 1 tejaldesai primarygroup 321 Mar 23 11:27 /tmp/echo.20180323112700
-rw-r--r-- 1 tejaldesai primarygroup 107 Mar 23 11:28 /tmp/echo.20180323112800

```
3. After one minute and more, both files are cleaned up.
```
$bazel run demo/ctc:ctc_demo -- 
echo
$ ls -al /tmp/echo*
lrwxrwxrwx 1 tejaldesai primarygroup  24 Mar 23 11:29 /tmp/echo -> /tmp/echo.20180323112900
-rw-r--r-- 1 tejaldesai primarygroup 214 Mar 23 11:29 /tmp/echo.20180323112900
```